float64[3] u
